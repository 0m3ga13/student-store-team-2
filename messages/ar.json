{
    "Index": {
        "Title": "المتجر الصغير",
        "Home": "الرئيسية",
        "Products": "المنتجات",
        "Orders": "الطلبيات",
        "State": "الحالة",
        "Buy": "شراء",
        "Borrow": "إعارة",
        "Description": "هذه المنصة هي مربحة للجميع: القرضاء يحققون دخلاً إضافيًا، والمقترضون يقللون التكاليف.",
        "Profile": "الحسـاب",
        "For Sale": "للبيع",
        "For Borrow": "للإعارة",
        "Sale & Borrow": "إعارة وشراء",
        "Day": "لليوم",
        "How It Works": "طريـقة العمل",
        "Hello": "مرحبا",
        "Sign out": "تسجيل الخروج",
        "Sign In": "تسجيل الدخول",
        "Login with Google": "سحل الدخول بحسابك",
        "Sign Up With Google": "سحل الدخول بحسابك"
    }
}
