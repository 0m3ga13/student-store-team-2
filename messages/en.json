{
    "Index": {
        "Title": "MiniStore",
        "Home": "Home",
        "Products": "Products",
        "Orders": "Orders",
        "State": "State",
        "Donate": "Donate",
        "Buy": "Buy",
        "Borrow": "Borrow",
        "Description": "This platform is a win-win for everyone involved lenders earn extra income, borrowers cut costs.",
        "Profile": "Profile",
        "for_sale": "for_sale",
        "for_borrow": "for_borrow",
        "Sale & Borrow": "Sale & Borrow",
        "Day": "Day",
        "How It Works": "How It Works",
        "recently added": "recently added",
        "Hello": "Hello",
        "Sign out": "Sign out",
        "Sign In": "Sign In",
        "Login with Google": "Login with Google",
        "Sign Up With Google": "Sign Up With Google",
        "Free delivery": "Free delivery",
        "feature descriprion 1": "You can receive your order without incurring any additional shipping costs",
        "qualite guarantee": "qualite guarantee",
        "feature descriprion 2": "Our products or services meet high standards and perform as promised.",
        "daily offers": "daily offers",
        "feature descriprion 3": "Exclusive discounts and promotions on a daily basis",
        "100% secure payement": "100% secure payement",
        "feature descriprion 4": "Your financial information is protected and your transactions are safe from unauthorized access",
        "My Dashboard": "My Dashboard",
        "Awaiting Pickup": "Awaiting Pickup",
        "All": "All",
        "Active": "Active",
        "Completed": "Completed",
        "Sort": "Sort:",
        "Newest": "Newest",
        "Oldest": "Oldest",
        "my stuff":"my stuff",
        "borrowings": "borrowings",
        "lendings": "lendings",
        "purchases": "purchases",
        "sales": "sales",
        "My Borrowings": "My Borrowings",
        "My Lendings": "My Lendings",
        "My Purchases": "My Purchases",
        "My Sales": "My Sales",
        "General details": "General details",
        "First Name": "First Name",
        "Last Name": "Last Name",
        "Birth Date": "Birth Date",
        "Gender": "Gender",
        "Institution": "Institution",
        "Contact & address Detail": "Contact & address Detail",
        "Phone Number": "Phone Number",
        "Building": "Building",
        "Street": "Street",
        "City": "City",
        "State_place": "State",
        "Save Changes": "Save Changes",
        "Upload Image": "Upload Image",
        "Close": "Close",
        "Upload": "Upload",
        "Loading": "Loading",
        "Not a member ?": "Not a member ?",
        "Sign Up": "Sign Up",
        "Store": "Store",
        "Create Your": "Create Your",
        "and start exchanging with other students": "and start exchanging with other students",
        "Or Login Using:": "Or Login Using:",
        "Login": "Login",
        "LOGIN": "LOGIN ",
        "Already a member ?": "Already a member ?",
        "Email and password are required": "Email and password are required",
        "Password should be at least 8 characters long": "Password should be at least 8 characters long",
        "and Conditions and Privacy Policy of The website": "and Conditions and Privacy Policy of The website",
        "Email": "Email",
        "Password": "Password",
        "Full Name": "Full Name",
        "Address": "Address",
        "and Privacy Policy": "and Privacy Policy",
        "I have read and agree with all of the Terms": "I have read and agree with all of the Terms",
        "PROFILE": "PROFILE",
        "SECURITY": "SECURITY",
        "Upload ID": "Upload ID",
        "Edit ID": "Edit ID",
        "Edit Image": "Edit Image",
        "WHY DONATE": "WHY DONATE",
        "MAKE A DIFFERENCE BY DONATING": "MAKE A DIFFERENCE BY DONATING",
        "Donate paragraph": "In our fast-paced lives, it's easy to forget the power of gratitude. Students, often on tight budgets, rely on our support. Your donation is more than just financial aid; it's a symbol of gratitude. By giving generously, you're helping students create a brighter future, one contribution at a time.",
        "Small Help": "Small Help",
        "Some Help": "Some Help",
        "Big Help": "Big Help",
        "List an Item": "List an Item",
        "Drag and drop some files here or click to select files": "Drag and drop some files here or click to select files",
        "Location": "Location",
        "Category": "Category",
        "books": "books",
        "kitchen": "kitchen",
        "sports": "sports",
        "games": "games",
        "clothes": "clothes",
        "electronics": "electronics",
        "furniture": "furniture",
        "description": "Description",
        "Product Condition": "Product Condition",
        "Sell": "Sell",
        "Price Per Week": "Price Per Week",
        "Price": "Price",
        "Upload Image(s)": "Upload Image(s)",
        "List": "List",
        "Product Name": "Product Name",
        "Comments": "Comments",
        "Type your comment": "Type your comment",
        "Type your reply": "Type your reply",
        "Respond": "Respond",
        "Like": "Like",
        "Send": "Send",
        "Authentication error": "Invalid login credentials. Please check your email and password.",
        "Filter": "Filter",
        "Pending": "Pending",
        "Quick Links": "Quick Links",
        "About LACITÉ": "ABOUT LACITÉ",
        "About Us": "ABOUT US",
        "Github Project Link": "GITHUB PROJECT LINK",
        "Contact Us": "Contact Us",
        "We Couldn't Find the page you're looking for": "We Couldn't Find the page you're looking for",
        "Go To Home": "Go To Home",
        "We Couldn't Find the requested Product": "We Couldn't Find the requested Product."
    }
}
