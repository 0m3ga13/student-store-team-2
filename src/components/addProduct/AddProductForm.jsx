"use client";

import { addDoc, collection } from "firebase/firestore";
import Image from "next/image";
import { useCallback, useEffect, useState } from "react";
import { useDropzone } from "react-dropzone";
import { SiXamarin } from "react-icons/si";

import { getSignature } from "@/lib/_cloudinary";
import { db } from "@/lib/firebase";

const AddProductForm = ({ className, categories }) => {
    const [files, setFiles] = useState([]);
    const [productName, setProductName] = useState("");
    const [category, setCategory] = useState("");
    const [location, setLocation] = useState("");
    const [description, setDescription] = useState("");
    const [type, setType] = useState("");
    const [sellPrice, setSellPrice] = useState("");
    const [borrowPrice, setBorrowPrice] = useState("");

    const onDrop = useCallback((acceptedFiles) => {
        if (acceptedFiles?.length) {
            setFiles((previousFiles) => [
                ...previousFiles,
                ...acceptedFiles.map((file) =>
                    Object.assign(file, { preview: URL.createObjectURL(file) })
                ),
            ]);
        }
    }, []);

    const removeFile = (name, event) => {
        event.stopPropagation();
        const updatedFiles = files.filter((file) => file.name !== name);
        setFiles(updatedFiles);
    };

    const { getRootProps, getInputProps, isDragActive } = useDropzone({
        accept: "image/*",
        maxSize: 1024 * 1000,
        maxFiles: 4,
        onDrop,
    });

    useEffect(() => {
        // Revoke the data uris to avoid memory leaks
        return () => files.forEach((file) => URL.revokeObjectURL(file.preview));
    }, [files]);

    const addMoreImages = () => {
        document.querySelector('input[type="file"]').click();
    };

    const handleFileChange = (e) => {
        const fileList = e.target.files;
        const newFiles = Array.from(fileList);
        if (newFiles.length > 0) {
            setFiles([
                ...files,
                ...newFiles.map((file) =>
                    Object.assign(file, {
                        preview: URL.createObjectURL(file),
                    })
                ),
            ]);
        }
    };

    async function action() {
        const uploadedFiles = [];
        for (const file of files) {
            const { signature, timestamp } = await getSignature();

            const formData = new FormData();
            formData.append("file", file);
            formData.append(
                "api_key",
                process.env.NEXT_PUBLIC_CLOUDINARY_API_KEY
            );
            formData.append("signature", signature);
            formData.append("timestamp", timestamp);
            formData.append("folder", "next");

            const endpoint = process.env.NEXT_PUBLIC_CLOUDINARY_UPLOAD_URL;
            const response = await fetch(endpoint, {
                method: "POST",
                body: formData,
            }).then((res) => res.json());

            uploadedFiles.push({
                name: file.name,
                preview: URL.createObjectURL(file),
                cloudinaryUrl: response.secure_url,
            });
        }

        setFiles(uploadedFiles);
        const imageLinks = uploadedFiles.map((file) => file.cloudinaryUrl);
        const productData = {
            productName,
            category,
            location,
            description,
            type,
            sellPrice,
            borrowPrice,
            images: imageLinks,
        };
        const productsDataCollection = collection(db, "products");
        await addDoc(productsDataCollection, productData);

        setProductName("");
        setCategory("");
        setLocation("");
        setDescription("");
        setType("");
        setSellPrice("");
        setBorrowPrice("");
        setFiles([]);
    }

    const ImageFile = ({ file }) => (
        <>
            <Image
                src={file.preview}
                alt={file.name}
                width={100}
                height={100}
                className='h-full w-full rounded-md object-contain'
            />
            <button
                type='button'
                className='absolute -right-3 -top-3 flex h-7 w-7 items-center justify-center rounded-full border border-rose-400 bg-rose-400 transition-colors hover:bg-white'
                onClick={(event) => removeFile(file.name, event)}
            >
                <SiXamarin className='h-5 w-5 fill-white transition-colors hover:fill-rose-400' />
            </button>
        </>
    );
    const renderPriceInput = () => {
        if (type === "Sell") {
            return (
                <input
                    type='number'
                    className='w-full border border-accent/50 placeholder:text-accent/50 px-4 py-2 rounded-md'
                    placeholder='Price'
                    value={sellPrice}
                    onChange={(e) => setSellPrice(e.target.value)}
                />
            );
        } else if (type === "Borrow") {
            return (
                <input
                    type='number'
                    className='w-full border border-accent/50 placeholder:text-accent/50 px-4 py-2 rounded-md placeholder:text-[12px]'
                    placeholder='Borrow Per Week'
                    value={borrowPrice}
                    onChange={(e) => setBorrowPrice(e.target.value)}
                />
            );
        } else if (type === "Sell & Borrow") {
            return (
                <>
                    <input
                        type='number'
                        className='w-1/2 border border-accent/50 placeholder:text-accent/50 px-4 py-2 rounded-md placeholder:text-[12px]'
                        placeholder='Sell Price'
                        value={sellPrice}
                        onChange={(e) => setSellPrice(e.target.value)}
                    />
                    <input
                        type='number'
                        className='w-1/2 border border-accent/50 placeholder:text-accent/50 px-4 py-2 rounded-md placeholder:text-[12px]'
                        placeholder='Borrow Per Week'
                        value={borrowPrice}
                        onChange={(e) => setBorrowPrice(e.target.value)}
                    />
                </>
            );
        }
    };
    return (
        <form
            action={action}
            className='flex flex-row justify-center items-center space-x-8 '
        >
            <div className='w-96'>
                {files.length === 0 && (
                    <div
                        {...getRootProps({ className: className })}
                        className='border border-dashed border-accent/50 flex flex-col items-center justify-center text-center p-8 text-accent/50'
                    >
                        <input
                            {...getInputProps({ name: "file", multiple: true })}
                        />
                        <p>
                            Drag and drop some files here, or click to select
                            files
                        </p>
                    </div>
                )}

                {files.length === 4 && (
                    <div className='flex flex-col gap-4'>
                        <div className='w-full relative rounded-md shadow-lg'>
                            <ImageFile file={files[0]} />
                        </div>
                        <div className='flex '>
                            {files.slice(1).map((file) => (
                                <div
                                    key={file.name}
                                    className='w-1/3 mx-2 relative rounded-md shadow-lg'
                                >
                                    <ImageFile file={file} />
                                </div>
                            ))}
                        </div>
                    </div>
                )}

                {files.length === 3 && (
                    <div className='flex flex-col gap-4'>
                        <div className='w-full relative rounded-md shadow-lg'>
                            <ImageFile file={files[0]} />
                        </div>
                        <div className='flex '>
                            {files.slice(1).map((file) => (
                                <div
                                    key={file.name}
                                    className='w-1/2 mx-2 relative rounded-md shadow-lg'
                                >
                                    <ImageFile file={file} />
                                </div>
                            ))}
                        </div>
                    </div>
                )}

                {files.length > 0 && files.length < 3 && (
                    <div className='flex flex-wrap gap-4'>
                        {files.map((file) => (
                            <div
                                key={file.name}
                                className='h-32 w-32 relative rounded-md shadow-lg'
                            >
                                <ImageFile file={file} />
                            </div>
                        ))}
                    </div>
                )}
            </div>
            <div className='flex flex-wrap border border-accent/50 p-4 w-96 space-y-4 '>
                <input
                    type='text'
                    placeholder='Product Name'
                    className='w-full border border-accent/50 placeholder:text-accent/50 px-4 py-2 rounded-md'
                    value={productName}
                    onChange={(e) => setProductName(e.target.value)}
                />

                <div className='flex justify-center mx-auto space-x-2'>
                    <select
                        value={category}
                        onChange={(e) => setCategory(e.target.value)}
                        className='w-1/2 border border-accent/50  placeholder:text-accent/50 px-4 py-2 rounded-md'
                    >
                        <option value=''>Select Category</option>
                        {categories.map((cat) => (
                            <option key={cat.index} value={cat}>
                                {cat}
                            </option>
                        ))}
                    </select>
                    <input
                        type='text'
                        placeholder='Location'
                        className='w-1/2 border border-accent/50  placeholder:text-accent/50 px-4 py-2 rounded-md'
                        value={location}
                        onChange={(e) => setLocation(e.target.value)}
                    />
                </div>

                <textarea
                    placeholder='Description'
                    className='w-full py-12 border border-accent/50  placeholder:text-accent/50 px-4 py-2 rounded-md flex items-center justify-center placeholder:text-center'
                    value={description}
                    onChange={(e) => setDescription(e.target.value)}
                ></textarea>

                <div className='flex justify-center mx-auto space-x-2'>
                    <select
                        value={type}
                        className='w-1/2 border border-accent/50  placeholder:text-accent/50  py-2 rounded-md text-[12px]'
                        onChange={(e) => setType(e.target.value)}
                    >
                        <option value='Borrow'>Borrow</option>
                        <option value='Sell'>Sell</option>
                        <option value='Sell & Borrow'>Sell & Borrow</option>
                    </select>
                    {renderPriceInput()}
                </div>

                <div className='flex space-x-2 w-full '>
                    <input
                        type='file'
                        style={{ display: "none" }}
                        onChange={handleFileChange}
                        multiple
                    />
                    <button
                        type='button'
                        onClick={addMoreImages}
                        className='mt-2 rounded-md border border-accent px-3 py-2 text-[12px] font-bold uppercase tracking-wider text-white transition-colors hover:bg-titleContent/80 hover:text-white bg-titleContent'
                    >
                        Upload Image(s)
                    </button>
                    <button
                        type='submit'
                        className='mt-2 rounded-md  border border-accent px-3 py-2 text-[12px] font-bold uppercase tracking-wider text-white transition-colors  hover:bg-accent/80  hover:text-white bg-accent'
                        disabled={files.length === 0}
                    >
                        Submit
                    </button>
                </div>
            </div>
        </form>
    );
};

export default AddProductForm;
