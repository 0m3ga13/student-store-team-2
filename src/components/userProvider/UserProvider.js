"use client";

import { createContext, useContext, useEffect, useState } from "react";

const UserContext = createContext();

export function UserProvider({ children, getCurrentUser }) {
    const [user, setUser] = useState(null);
    const [userData, setUserData] = useState(null);
    const [loading, setLoading] = useState(true);
    useEffect(() => {
        const fetchData = async () => {
            try {
                const data = await getCurrentUser();
                setUser(data?.user);
            } catch (error) {
                console.error(error);
            }
        };

        fetchData();
    }, []);

    useEffect(() => {
        const fetchUserData = async () => {
            try {
                if (user) {
                    await setUserData(user.user_metadata);
                    setLoading(false);
                } else {
                    await new Promise((resolve) => setTimeout(resolve, 1000));

                    setLoading(false);
                }
            } catch (error) {
                // Handle errors
                setLoading(false);
                throw error;
            }
        };

        fetchUserData();
    }, [user]);

    return (
        <UserContext.Provider value={{ user, userData, loading }}>
            {children}
        </UserContext.Provider>
    );
}

export function useUser() {
    return useContext(UserContext);
}
