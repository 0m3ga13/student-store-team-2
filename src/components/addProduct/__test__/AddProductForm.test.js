import { NextIntlClientProvider } from "next-intl";
import renderer from "react-test-renderer";

import { UserProvider } from "@/components/userProvider/UserProvider";

import AddProductForm from "../AddProductForm";
import messages from "../../../../messages/en.json";

// Mock the "next/navigation" module
jest.mock("next/navigation", () => ({
    useRouter: jest.fn(() => ({
        push: jest.fn(),
    })),
    useSearchParams: jest.fn(() => ({
        get: jest.fn(),
    })),
    usePathname: jest.fn(),
    useParams: () => ({
        locale: "en",
    }),
}));

jest.mock("@supabase/supabase-js", () => {
    return {
        ...jest.requireActual("@supabase/supabase-js"), // Use the actual module for non-mocked parts
        createClient: jest.fn(() => ({
            from: jest.fn(() => ({
                select: jest.fn(() => ({
                    eq: jest.fn(() => ({
                        // Customize the behavior of your Supabase queries as needed
                        data: { id: 123, name: "John Doe" },
                        error: null,
                    })),
                })),
            })),
        })),
    };
});

const categories = [["Electronics", "Books", "Clothing"]];

it("renders correctly", async () => {
    const tree = renderer
        .create(
            <NextIntlClientProvider locale='en' messages={messages}>
                <UserProvider>
                    <AddProductForm categories={categories} />{" "}
                </UserProvider>
            </NextIntlClientProvider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
