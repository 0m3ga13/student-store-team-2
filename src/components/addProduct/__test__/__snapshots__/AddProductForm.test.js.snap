// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <form
    className="flex md:flex-row flex-col justify-center items-center space-x-8 sm:mb-0 mb-5 "
    onSubmit={[Function]}
  >
    <div
      className="md:w-1/2 hidden sm:block"
    >
      <div
        className="border border-dashed border-accent/50 md:flex flex flex-col items-center justify-center text-center p-8 text-accent/50 text-sm"
        onBlur={[Function]}
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="presentation"
        tabIndex={0}
      >
        <input
          multiple={true}
          name="file"
          onChange={[Function]}
          onClick={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
          tabIndex={-1}
          type="file"
        />
        <p>
          Drag and drop some files here or click to select files
        </p>
      </div>
    </div>
    <div
      className="flex flex-wrap items-center justify-center  p-4 md:w-1/2 space-y-4 text-lg max-w-full"
    >
      <input
        className="border border-accent/50 placeholder:text-accent/50 px-4 py-2 rounded-md w-full"
        onChange={[Function]}
        placeholder="Product Name"
        required={true}
        type="text"
        value=""
      />
      <select
        className="border border-accent/50  placeholder:text-accent/50 px-4 py-2 rounded-md w-1/2 uppercase"
        onChange={[Function]}
        required={true}
        value=""
      >
        <option
          value=""
        >
          Category
        </option>
        <option>
          Index
        </option>
      </select>
      <div
        className="relative mb-4"
      >
        <input
          className="border border-accent/50  placeholder:text-accent/50 px-4 py-2 rounded-md w-full uppercase"
          onChange={[Function]}
          placeholder="Enter Location"
          required={true}
          type="text"
          value=""
        />
      </div>
      <textarea
        className="w-full h-48 border border-accent/50  placeholder:text-accent/50 px-4  rounded-md placeholder:flex placeholder:items-center placeholder:justify-center placeholder:text-center placeholder:text-md"
        onChange={[Function]}
        placeholder="Description"
        required={true}
        value=""
      />
      <div
        className="flex justify-center mx-auto space-x-2 w-full"
      >
        <select
          className="w-1/2 border border-accent/50  placeholder:text-accent/50  py-2 rounded-md text-sm uppercase"
          onChange={[Function]}
          required={true}
          value="for_sale"
        >
          <option
            value="for_borrow"
          >
            Borrow
          </option>
          <option
            value="for_sale"
          >
            Sell
          </option>
        </select>
        <input
          className="w-full border border-accent/50 placeholder:text-accent/50 px-4 py-2 rounded-md"
          onChange={[Function]}
          placeholder="Price"
          type="number"
          value=""
        />
      </div>
      <div
        className="flex justify-center mx-auto space-x-2 items-center text-sm"
      >
        <label>
          Product Condition
          :
        </label>
        <select
          className="w-1/2 border border-accent/50 placeholder:text-accent/50 md:px-4 py-2 rounded-md "
          onChange={[Function]}
          required={true}
          value={5}
        >
          <option
            value={0}
          >
            0
            /10
          </option>
          <option
            value={1}
          >
            1
            /10
          </option>
          <option
            value={2}
          >
            2
            /10
          </option>
          <option
            value={3}
          >
            3
            /10
          </option>
          <option
            value={4}
          >
            4
            /10
          </option>
          <option
            value={5}
          >
            5
            /10
          </option>
          <option
            value={6}
          >
            6
            /10
          </option>
          <option
            value={7}
          >
            7
            /10
          </option>
          <option
            value={8}
          >
            8
            /10
          </option>
          <option
            value={9}
          >
            9
            /10
          </option>
          <option
            value={10}
          >
            10
            /10
          </option>
        </select>
      </div>
      <input
        className="border border-accent/50 placeholder:text-accent/50 px-4 py-2 rounded-md w-full"
        onChange={[Function]}
        placeholder="Additional Information"
        type="text"
        value=""
      />
      <div
        className="flex space-x-2 w-full justify-center"
      >
        <input
          multiple={true}
          onChange={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
          type="file"
        />
        <button
          className="mt-2 rounded-md border border-accent px-4 py-2 text-sm font-bold uppercase tracking-wider text-white transition-colors hover:bg-titleContent/80 hover:text-white bg-titleContent disabled:bg-titleContent/20 disabled:border-none"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Upload Image(s)
        </button>
        <button
          className="mt-2 rounded-md  border border-accent px-4 py-2 text-sm font-bold uppercase tracking-wider text-white transition-colors  hover:bg-accent/80  hover:text-white bg-accent disabled:bg-accent/20 disabled:border-none"
          disabled={true}
          type="submit"
        >
          List
        </button>
      </div>
    </div>
    <div
      className="sm:hidden flex flex-wrap w-full justify-center "
    />
  </form>,
  " ",
]
`;
